app/src/main/java/org/mozilla/fenix/tabtray/TabTrayView.kt

```diff
@@ -239,6 +240,31 @@ class TabTrayView(
         }
 
         adjustNewTabButtonsForNormalMode()
+
+        val shouldShowCFR =
+            view.context.settings().canShowCfr &&
+            tabs.size >= OPEN_TAB_CFR_AMOUNT &&
+            view.context.settings().shouldShowAutoCloseTabsBanner
+
+        view.context.settings().shouldShowAutoCloseTabsBanner = true
+
+        if (shouldShowCFR) {
+            InfoBanner(
+                context = view.context,
+                message = view.context.getString(R.string.tab_tray_close_tabs_banner_message),
+                dismissText = view.context.getString(R.string.tab_tray_close_tabs_banner_negative_button_text),
+                actionText = view.context.getString(R.string.tab_tray_close_tabs_banner_positive_button_text),
+                container = view.infoBanner,
+                dismissByHiding = true,
+                dismissAction = { view.context.settings().shouldShowAutoCloseTabsBanner = false }
+            ) {
+                view.context.settings().shouldShowAutoCloseTabsBanner = false
+                interactor.onSetUpAutoCloseTabsClicked()
+            }.apply {
+                view.infoBanner.visibility = View.VISIBLE
+                showBanner()
```

Comment:
Let's move the reseting of the cfr shown time (`settings.lastCfrShownTimeInMillis = System.currentTimeMillis()`) for this banner and from `OpenInAppOnboardingObserver` into the `showBanner` fun?